### PR TITLE
IGNITE-13968 Add snapshot events documentation pages

### PR DESCRIPTION
**Pull request by Mmuzaf**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: docs/_docs/events/events.adoc [hunk 1]
```diff
@@ -325,6 +325,22 @@ They allow you to get notification about different stages of transaction executi
 | EVT_TX_RESUMED | A transaction is resumed. |The node where the transaction was started.
 |===
 
+== Snapshot Events
+
+All snapshot events are instances of the link:{events_url}/SnapshotEvent.html[SnapshotEvent] class. The snapshot events
+always triggered at the node initiated a snapshot operation and associate with the particular snapshot operation name.
```
Comment (Nikita-tech-writer) [inline docs/_docs/events/events.adoc:331]:
```suggestion
are always triggered at the node initiated a snapshot operation. They are also associated with the particular snapshot operation name.
```

File: docs/_docs/events/events.adoc [hunk 2]
```diff
@@ -325,6 +325,22 @@ They allow you to get notification about different stages of transaction executi
...
+These events allow you to be notified about the states and result of the cluster snapshot operation for further processing
```
Comment (Nikita-tech-writer) [inline docs/_docs/events/events.adoc:332]:
```suggestion
These events can notify you about the states and the result of the cluster snapshot operation, which can be used for further processing in
```

File: docs/_docs/events/events.adoc [hunk 3]
```diff
@@ -325,6 +325,22 @@ They allow you to get notification about different stages of transaction executi
...
+your custom procedures.
+
+
+[cols="2,5,3",opts="header"]
+|===
+| Event Type | Event Description | Where Event Is Fired
```
Comment (Nikita-tech-writer) [inline docs/_docs/events/events.adoc:338]:
```suggestion
| Event Type | Event Description | Where Event is Fired
```

File: docs/_docs/events/events.adoc [hunk 4]
```diff
@@ -325,6 +325,22 @@ They allow you to get notification about different stages of transaction executi
...
+| EVT_CLUSTER_SNAPSHOT_STARTED | A cluster snapshot operation is started. | The node where the snapshot operation was initiated.
+| EVT_CLUSTER_SNAPSHOT_FINISHED | A cluster snapshot operation is finished. The event fired when the snapshot operation finished on all the cluster nodes. |  The node where the snapshot operation was finished.
```
Comment (Nikita-tech-writer) [inline docs/_docs/events/events.adoc:340]:
```suggestion
| EVT_CLUSTER_SNAPSHOT_FINISHED | A cluster snapshot operation is finished. The event is fired when the snapshot operation is finished on all the cluster nodes. |  The node where the snapshot operation was finished.
```

File: docs/_docs/events/events.adoc [hunk 5]
```diff
@@ -325,6 +325,22 @@ They allow you to get notification about different stages of transaction executi
...
+| EVT_CLUSTER_SNAPSHOT_FAILED | A cluster snapshot operation is failed. | The node where the snapshot operation was initiated. Note that message will contain the short reason of the failure from another node. You may check the logs for details.
```
Comment (Nikita-tech-writer) [inline docs/_docs/events/events.adoc:341]:
```suggestion
| EVT_CLUSTER_SNAPSHOT_FAILED | A cluster snapshot operation is failed. | The node where the snapshot operation was initiated. Note that the message will contain a failure description from another node. You may check the logs for details.
```

**Review by Nikita-tech-writer — CHANGES_REQUESTED** [review]
Please see my suggested changes to the PR.
Feel free to comment and discuss them.

**Comment by Mmuzaf** [comment]
@Nikita-tech-writer 

Thank you for the review. I've applied all your suggestions.

**Pull request merged**